src/sage/schemes/elliptic_curves/ell_finite_field.py

```diff
@@ -1436,6 +1436,7 @@ def has_order(self, value, num_checks=8):
             if not (value * G).is_zero():
                 return False
 
+        self._order = value
```

Comment:
Just to make sure, there is no path to this line that is impacted by the false positive bug? I.e. if this line is reached, the order is 100% equal to `value`?

Reply:
No, it's the other way around I think. It's possible for the code to reach this line with a wrong value.

Reply:
I think this can be reached in the bugged case. I think you were right before, let's not set it here for now until the issues are sorted out.

Reply:
> No, it's the other way around I think. It's possible for the code to reach this line with a wrong value.

Let's not set `order` here then, but maybe add a `TODO` comment to do so in the future when/if the bugs are sorted out.